pep-0596.rst

```diff
@@ -0,0 +1,191 @@
+PEP: 596
+Title: Use UTF-8 for default text file encoding
+Author: Inada Naoki  <songofacandy@gmail.com>
+Status: Draft
+Type: Standards Track
+Content-Type: text/x-rst
+Created: 04-Jun-2019
+Python-Version: 3.9
+
+Abstract
+========
+
+Currently, ``TextIOWrapper`` uses ``locale.getpreferredencoding(False)``
+when ``encoding`` is not specified.
+
+This PEP proposes changing the default encoding to "UTF-8" regardless of
+platform or locale.
+
+
+Motivation
+==========
+
+People assume UTF-8 always
+--------------------------
+
+Package authors using macOS or Linux may forget that the default encoding
+is not always UTF-8.
+
+For example, ``long_description = open("README.md").read()`` in
+``setup.py`` is common mistake.  If there are at least one emoji or any
+other non-ASCII characters in the ``README.md`` file, many Windows users
+cannot install the package by ``UnicodeDecodeError``.
+
+
+Code page is not stable
+-----------------------
+
+Some tools on Windows change code page to 65001 (UTF-8), and Microsoft
+is using UTF-8 and cp65001 more widely in recent Windows 10.
+
+For example, "Command Prompt" uses legacy code page by default.
+But WSL changes the code page to 65001, and  ``python.exe`` on Windows
+can be executed from WSL.  So ``python.exe`` executed from legacy
+console and from WSL cannot read text files written by each other.
+
+Many Windows users don't understand code pages and which code page is
+currently used.  So changing default text file encoding based on current
+code page may cause confusion.
+
+Consistent default text encoding will make Python behavior more expectable
+and easy to learn.
+
+
+Use UTF-8 by default is easier to new programmers
+-------------------------------------------------
+
+Python is one of the most popular first programming language.
+
+New programmers may not know about encoding.  When they download text data
+written in UTF-8 from the internet, they are forced to know encoding.
+
+Popular text editers like VS Code or Atom use UTF-8 by default.
+Even notepad.exe uses UTF-8 by default from Windows 10 2019 may update.
+(Note that Python 3.9 will be released in 2021.)
+
+It would be nice if new programmers are not forced to know about encoding
+until they need to handle text files encoded in encoding other than UTF-8.
+
+
+Specification
+=============
+
+From Python 3.9, default encoding of ``TextIOWrapper`` and ``open()`` is
+changed from ``locale.getpreferredencoding(False)`` to "UTF-8".
+
+When there is device encoding (``os.device_encoding(buffer.fileno())``),
+itstill precedes than this default encoding.
+
+
+Not affected areas
+------------------
+
+Unlike UTF-8 mode, ``locale.getpreferredencoding(False)`` returns
+locale encoding.  Locale encoding is still respected when it makes sense.
+
+``stdin``, ``stdout``, and ``stderr`` keep respecting locale too.  For example,
+these commands don't cause mojibake regardless code page::
```

Comment:
That sounds like a high risk of mojibake :-(

Let's say that a file contains a filename. Like a "Makefile" file.

`cat Makefile` and `python3.9 -c 'print(open("Makefile").read(), end="")'` will now output different bytes. cat reads bytes and writes unmodified bytes. python3.9 will decodes from UTF-8 and encodes to the locale encoding. Say hello to mojibake!

Reply:
It is impossible to avoid 100% of mojibake.
Even `cat Makefile` cause mojibake when Makefile contains UTF-8 but console uses legacy encoding.

In your example, it doesn't cause mojibake on Windows legacy console, thanks to PEP 528.